xbmc/utils/EGLUtils.cpp

```diff
@@ -667,13 +667,16 @@ void CEGLContextUtils::SetDamagedRegions(const CDirtyRegionList& dirtyRegions)
   }
   else
   {
-    EGLint height = eglQuerySurface(m_eglDisplay, m_eglSurface, EGL_HEIGHT, &height);
+    EGLint height = 1080;
+    eglQuerySurface(m_eglDisplay, m_eglSurface, EGL_HEIGHT, &height);
```

Comment:
What would practically happen if the call fails and region.y2 is larger than 1080?

Reply:
The UI would be unusable.

Reply:
Yeah. That was my hint ;-). So should we go with height of 1080 or should we rather make sure that the rect's height we set below is positiv?

Reply:
You really don't trust the EGL implementations out there, do you 🙂?

If the driver fails us here, we would need to return a bool that it did and subsequently draw the UI at full resolution, not just the subset which was suitable for the regions. In this case, we would essentially fall back to the default `<algorithmdirtyregions>3</algorithmdirtyregions>`. 

I don't think that adding such workarounds for broken embedded drivers is really worth it, since it also adds to the main application code. Because this code is gated behind an advanced setting (which has driver warnings in the wiki), the impact of a malfunctioning driver is not critical.

Or do you see a way this could fail in normal operation?

Reply:
In fact since I started with Kodi there was always a broken driver ;-). One broken differently than the other. Let's go with what we have now here. Reality will tell.

What you could do to help us track that issue: if it returns false log a debug